### ISSUE TITLE
The new representation of EO breaks all dependent components.

### ISSUE THREAD

**Issue by volodya-lombrozo** [objectionary/eo#2758]
I used to get rather good EO representation of `XMIR`:
```eo
seq > @
  tuple
    $
    .a
    .foo
    .plus
      1
```

With the `0.34.2` version of `eo-parser` I get the following output:
```
seq > @
      tuple
        plus.
          foo.
            a.
              $
          1
```
Which is extremely hard to read (I have to read bottom -> top to unveil the logic.) Also we previously have understandable literals:
```
"java/lang/Object" > supername
``` 
Now it's a complete mess:
```
  """
6A 61 76 61 2F 6C 61 6E 67 2F 4F 62 6A 65 63 74
""" > supername
```
So I see two critical problems here:
1. All the modules dependent on `eo-parser` are broken. One of the use cases of the previous representation involves tests. Therefore, to upgrade to `eo-parser:0.34.2`, I must upgrade all the tests, that use `EO` representation. It is extremely tedious task.
2. The new representation is designed for programs, not humans.
____

However, I understand why we might need to modify the EO output. Thus, I suggest returning to the previous representation and adding one more representation as a separate new class. This will repair all the dependent modules and you will be able to use new representation for other needs.

**Comment by volodya-lombrozo** [comment]
@yegor256 @maxonfjvipon Could you take a look, please?

**Comment by yegor256** [comment]
@maxonfjvipon I think it's a bug:

```
  """
6A 61 76 61 2F 6C 61 6E 67 2F 4F 62 6A 65 63 74
""" > supername
```

**Comment by maxonfjvipon** [comment]
@volodya-lombrozo could you please provide an XMIR you get the EO output from?

**Comment by volodya-lombrozo** [comment]
@maxonfjvipon of course: 
```xml
<?xml version="1.0" encoding="UTF-8"?>
<program dob="2023-12-29T11:59:49.384058Z"
         ms="1703851189384"
         name="j$Simple"
         revision="0.0.0"
         time="2023-12-29T11:59:49.384058Z"
         version="0.0.0">
   <listing>yv66vgAAADcAFwoAAgADBwAEDAAFAAYBABBqYXZhL2xhbmcvT2JqZWN0AQAGPGluaXQ+AQADKClWBwAIAQAXamF2YS9sYW5nL1N0cmluZ0J1aWxkZXIIAAoBAANhYmMKAAcADAwABQANAQAVKExqYXZhL2xhbmcvU3RyaW5nOylWCgAHAA8MABAAEQEABmFwcGVuZAEAHChJKUxqYXZhL2xhbmcvU3RyaW5nQnVpbGRlcjsHABMBAAZTaW1wbGUBAARDb2RlAQAEbWFpbgEAFihbTGphdmEvbGFuZy9TdHJpbmc7KVYAIQASAAIAAAAAAAIAAQAFAAYAAQAUAAAAEQABAAEAAAAFKrcAAbEAAAAAAAkAFQAWAAEAFAAAABsAAwABAAAAD7sAB1kSCbcACwS2AA5XsQAAAAAAAA==</listing>
   <errors/>
   <sheets/>
   <license/>
   <metas>
      <meta>
         <head>package</head>
         <tail/>
         <part/>
      </meta>
      <meta>
         <head>alias</head>
         <tail>org.eolang.jeo.opcode</tail>
         <part>org.eolang.jeo.opcode</part>
      </meta>
      <meta>
         <head>alias</head>
         <tail>org.eolang.jeo.label</tail>
         <part>org.eolang.jeo.label</part>
      </meta>
   </metas>
   <objects>
      <o abstract="" name="j$Simple">
         <o base="int" data="bytes" name="access">00 00 00 00 00 00 00 21</o>
         <o base="string" data="bytes" name="supername">6A 61 76 61 2F 6C 61 6E 67 2F 4F 62 6A 65 63 74</o>
         <o base="tuple" data="tuple" name="interfaces"/>
         <o abstract="" name="new">
            <o base="int" data="bytes" name="access">00 00 00 00 00 00 00 01</o>
            <o base="string" data="bytes" name="descriptor">28 29 56</o>
            <o base="string" data="bytes" name="signature"/>
            <o base="tuple" data="tuple" name="exceptions"/>
            <o base="seq" name="@">
               <o base="tuple">
                  <o base=".super">
                     <o base="$"/>
                  </o>
                  <o base="opcode" name="RETURN">
                     <o base="int" data="bytes">00 00 00 00 00 00 00 B1</o>
                  </o>
               </o>
            </o>
            <o base="tuple" name="trycatchblocks"/>
         </o>
         <o abstract="" name="j$main">
            <o base="int" data="bytes" name="access">00 00 00 00 00 00 00 09</o>
            <o base="string" data="bytes" name="descriptor">28 5B 4C 6A 61 76 61 2F 6C 61 6E 67 2F 53 74 72 69 6E 67 3B 29 56</o>
            <o base="string" data="bytes" name="signature"/>
            <o base="tuple" data="tuple" name="exceptions"/>
            <o abstract="" name="arg__[Ljava/lang/String;__0"/>
            <o base="seq" name="@">
               <o base="tuple">
                  <o base=".append">
                     <o base=".new">
                        <o base="java/lang/StringBuilder"/>
                        <o base="string" data="bytes">61 62 63</o>
                     </o>
                     <o base="int" data="bytes">00 00 00 00 00 00 00 01</o>
                  </o>
                  <o base="opcode" name="RETURN">
                     <o base="int" data="bytes">00 00 00 00 00 00 00 B1</o>
                  </o>
               </o>
            </o>
            <o base="tuple" name="trycatchblocks"/>
         </o>
      </o>
   </objects>
</program>
```

**Comment by support512** [comment]
> The new representation is designed for programs, not humans.

Exactly, it's like everything else the author of this repo does. He provides the service to the computer, not people 😂 (anyway, it's interesting to follow, thanks, Yegor)

**Comment by maxonfjvipon** [comment]
The decision was made:
1. Extend EO syntax so it's allowed to use vertical methods as application arguments:
```
x
  a
  .b
  .c:y

# The same as

x a.b.c:y

# Or

x
  a.b.c:y
```

2. `PrintMojo` and `XMIR` class should print methods in vertical notation, not in reversed

**Comment by volodya-lombrozo** [comment]
Maybe it's better to have several options how to represent eo? What do you think?

**Comment by maxonfjvipon** [comment]
@volodya-lombrozo maybe you're right, reversed printing is already done and works, so we can leave it

**Comment by 0pdd** [comment]
@volodya-lombrozo the puzzle [#2786](https://github.com/objectionary/eo/issues/2786) is still not solved; solved: [#2787](https://github.com/objectionary/eo/issues/2787).

**Comment by yegor256** [comment]
@rultor release, tag is `0.35.0`

**Comment by maxonfjvipon** [comment]
@volodya-lombrozo please update EO version and try to use new `Xmir.Default` object to print xmir to EO in "strait" notation. To print in "reversed" notation you can use `XmirReversed` object.
Let me know if result EO is good enough for you

**Comment by volodya-lombrozo** [comment]
@maxonfjvipon https://github.com/objectionary/eo/issues/2801 I faced the problem

**Comment by volodya-lombrozo** [comment]
I have two concerns about the current implementation: #2801 #2802 
But it is better to continue the discussion in the relevant issues.
So I close the current issue.

**Comment by 0pdd** [comment]
@volodya-lombrozo all 2 puzzles are solved here: [#2786](https://github.com/objectionary/eo/issues/2786), [#2787](https://github.com/objectionary/eo/issues/2787).
